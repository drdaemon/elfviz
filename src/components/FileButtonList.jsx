import React from 'react';
import PropTypes from 'prop-types';

const FileButtonList = ({ files, onClick }) => {
  const filenames = Object.keys(files);
  const links = filenames.map((filename, index) => (
    <a
      href="#"
      key={'file-' + filename}
      className="btn btn-xs btn-default"
      onClick={onClick.bind(this, filename, files[filename])}
      title={filename + '.json'}
    >
      {index + 1}
    </a>
  ));
  return <span>{links}</span>;
};

FileButtonList.propTypes = {
  files: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default FileButtonList;
