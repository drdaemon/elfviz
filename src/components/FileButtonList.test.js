import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FileButtonList from './FileButtonList';

const files = {
  file1: {
    name: 'name1',
  },
  file2: {
    name: 'name2',
  },
}

describe('FileButtonList', () => {
  it('renders without crashing', () => {
    shallow(<FileButtonList files={{}} onClick={() => {}} />);
  });

  it('should render as span element', () => {
    const wrapper = shallow(<FileButtonList files={{}} onClick={() => {}} />);
    expect(wrapper.type()).toEqual('span');
  });

  it('should match an empty snapshot', () => {
    const wrapper = shallow(<FileButtonList files={{}} onClick={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot with files', () => {
    const wrapper = shallow(<FileButtonList files={files} onClick={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
