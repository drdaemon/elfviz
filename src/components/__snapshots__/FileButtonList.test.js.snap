// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileButtonList should match an empty snapshot 1`] = `<span />`;

exports[`FileButtonList should match snapshot with files 1`] = `
<span>
  <a
    className="btn btn-xs btn-default"
    href="#"
    onClick={[Function]}
    title="file1.json"
  >
    1
  </a>
  <a
    className="btn btn-xs btn-default"
    href="#"
    onClick={[Function]}
    title="file2.json"
  >
    2
  </a>
</span>
`;
